react-carousel-component/my-app/src/Carousel.js

```diff
@@ -0,0 +1,99 @@
+import { useState, useCallback, useEffect } from 'react';
+import {
+  FaChevronLeft,
+  FaChevronRight,
+  FaCircle,
+  FaRegCircle,
+} from 'react-icons/fa';
+
+export default function Carousel({ images }) {
+  const [currentIndex, setCurrentIndex] = useState(0);
+
+  function handleNextClick() {
+    setCurrentIndex((previousIndex) => (previousIndex + 1) % images.length);
+  }
+
+  function handlePreviousClick() {
+    setCurrentIndex(
+      (previousIndex) => (previousIndex - 1 + images.length) % images.length
+    );
+  }
+
+  function handleIndicatorClick(index) {
+    setCurrentIndex(index);
+  }
+
+  const nextImage = useCallback(() => {
+    setCurrentIndex((currentIndex + 1) % images.length);
+  }, [currentIndex, images]);
```

Comment:
I always debate on these: Do I make the dependency `images.length` or just `images`. I think the answer comes from the question, "What would happen if I got new images but the same number of them so `.length` doesn't change?"

In this case (but certainly not in all cases), it looks like it would continue operating correctly, so `images.length` would be the best answer. However, it will still work correctly if you use `images`.